protocol/storage/miner.go

```diff
@@ -706,70 +700,61 @@ func (sm *Miner) OnNewHeaviestTipSet(ts types.TipSet) {
 		return
 	}
 
-	provingPeriodStart, err := sm.getProvingPeriodStart()
+	provingPeriodStart, provingPeriodEnd, err := sm.getProvingPeriod()
 	if err != nil {
-		log.Errorf("failed to get provingPeriodStart: %s", err)
+		sm.log.Errorf("failed to get proving period: %s", err)
 		return
 	}
 
 	sm.postInProcessLk.Lock()
 	defer sm.postInProcessLk.Unlock()
 
-	if sm.postInProcess != nil && sm.postInProcess.Equal(provingPeriodStart) {
+	if sm.postInProcess != nil && sm.postInProcess.Equal(provingPeriodEnd) {
 		// post is already being generated for this period, nothing to do
 		return
 	}
 
 	height, err := ts.Height()
 	if err != nil {
-		log.Errorf("failed to get block height: %s", err)
-		return
-	}
-
-	sectorSize, err := sm.porcelainAPI.MinerGetSectorSize(ctx, sm.minerAddr)
-	if err != nil {
-		log.Errorf("failed to get miner's sector size: %s", err)
+		sm.log.Errorf("failed to get block height: %s", err)
 		return
 	}
 
 	// the block height of the new heaviest tipset
 	h := types.NewBlockHeight(height)
 
-	// compute the block height at which the miner's current proving period ends
-	provingPeriodEnd := provingPeriodStart.Add(types.NewBlockHeight(miner.ProvingPeriodDuration(sectorSize)))
-
 	if h.GreaterEqual(provingPeriodStart) {
 		if h.LessThan(provingPeriodEnd) {
```

Comment:
BLOCKING: There is a bug here - not one which you introduced. I think that if the miner submits their PoSt when the block height _equals_ the end of the proving period that their submission should not be penalized. Do you disagree?

Reply:
A PoSt is going to take a long time to generate, so I guess this doesn't matter.

Reply:
I think the ProvingPeriodEnd should be exclusive so proving periods don't overlap. The spec seems to disagree with me. As you mention here, it's mostly irrelevant, because it's very unlikely that PoSt generation could start on the last block of the proving period and make it on chain in the same block. I'll change it anyway.